debugtools/DDR_VM/src/com/ibm/j9ddr/AuxFieldInfo29.dat

```diff
@@ -451,6 +451,8 @@ J9ROMClass.specialSplitMethodRefIndexes = required
 J9ROMClass.staticSplitMethodRefCount = required
 J9ROMClass.staticSplitMethodRefIndexes = required
 J9ROMClass.superclassName = required
+J9ROMClass.varHandleMethodTypeCount = U32
+J9ROMClass.varHandleMethodTypeLookupTable = J9SRP
```

Comment:
For JDK17+, the below test fails by just adding the above two fields. In JDK17+, these fields don't exist in the `J9ROMClass`. re https://github.com/eclipse-openj9/openj9/pull/19001#discussion_r1499957186: Is there a way to ignore these fields? or do we need to use reflection?
```
     [java] [AutoRun] [INFO] Starting testDumpROMClassExt(j9vm.test.ddrext.junit.TestClassExt)
     ...
     [java] [DDRExtTesterBase] [ERROR] '!queryromclass 0x00007F58C5407000,/romHeader' output does not contain success key : romSize
     [java] [DDRExtTesterBase] [ERROR] '!queryromclass 0x00007F58C5407000,/romHeader' output :ROM Class 'java/lang/Object' at 0x00007F58C5407000:
     [java]
     [java]
     [java] === Section Start: romHeader (176 bytes)                                                  ===
     [java] 0x00007F58C5407000-0x00007F58C5407004 [           0x00000838 varHandleMethodTypeLookupTable ]  -> 0x00007F58C5407838
     [java] 0x00007F58C5407004-0x00007F58C5407008 [           0x00000000 singleScalarStaticCount      ]
     [java] 0x00007F58C5407008-0x00007F58C540700C [           0x00000510 className                    ]  -> 0x00007F58C5407518 java/lang/Object
     [java] 0x00007F58C540700C-0x00007F58C5407010 [           0x00000000 superclassName               ]
     [java] 0x00007F58C5407010-0x00007F58C5407014 [           0x00000021 modifiers                    ]
     [java] 0x00007F58C5407014-0x00007F58C5407018 [           0x08A00000 extraModifiers               ]
     [java] 0x00007F58C5407018-0x00007F58C540701C [           0x00000000 interfaceCount               ]
     [java] 0x00007F58C540701C-0x00007F58C5407020 [           0x0000016C interfaces                   ]  -> 0x00007F58C5407188
     [java] 0x00007F58C5407020-0x00007F58C5407024 [           0x0000000E romMethodCount               ]
     [java] 0x00007F58C5407024-0x00007F58C5407028 [           0x000001D4 romMethods                   ]  -> 0x00007F58C54071F8
     [java] 0x00007F58C5407028-0x00007F58C540702C [           0x00000000 romFieldCount                ]
     [java] 0x00007F58C540702C-0x00007F58C5407030 [           0x0000015C romFields                    ]  -> 0x00007F58C5407188
     [java] 0x00007F58C5407030-0x00007F58C5407034 [           0x00000000 objectStaticCount            ]
     [java] 0x00007F58C5407034-0x00007F58C5407038 [           0x00000000 doubleScalarStaticCount      ]
     [java] 0x00007F58C5407038-0x00007F58C540703C [           0x0000001B ramConstantPoolCount         ]
     [java] 0x00007F58C540703C-0x00007F58C5407040 [           0x0000001B romConstantPoolCount         ]
     [java] 0x00007F58C5407040-0x00007F58C5407048 [   0xFFFFFFFFFFFFFFC0 intermediateClassData        ]  -> 0x00007F58C5407000
     [java] 0x00007F58C5407048-0x00007F58C540704C [           0x00000838 intermediateClassDataLength  ]
     [java] 0x00007F58C540704C-0x00007F58C5407050 [           0x0000000E instanceShape                ]
     [java] 0x00007F58C5407050-0x00007F58C5407054 [           0x00000400 cpShapeDescription           ]  -> 0x00007F58C5407450
     [java] 0x00007F58C5407054-0x00007F58C5407058 [           0x00000000 outerClassName               ]
     [java] 0x00007F58C5407058-0x00007F58C540705C [           0x00000000 memberAccessFlags            ]
     [java] 0x00007F58C540705C-0x00007F58C5407060 [           0x00000000 innerClassCount              ]
     [java] 0x00007F58C5407060-0x00007F58C5407064 [           0x00000128 innerClasses                 ]  -> 0x00007F58C5407188
     [java] 0x00007F58C5407064-0x00007F58C5407068 [           0x00000000 enclosedInnerClassCount      ]
     [java] 0x00007F58C5407068-0x00007F58C540706C [           0x00000120 enclosedInnerClasses         ]  -> 0x00007F58C5407188
     [java] 0x00007F58C540706C-0x00007F58C5407070 [           0x00000000 nestHost                     ]
     [java] 0x00007F58C5407070-0x00007F58C5407072 [               0x0000 nestMemberCount              ]
     [java] 0x00007F58C5407072-0x00007F58C5407074 [               0x0000 unused                       ]
     [java] 0x00007F58C5407074-0x00007F58C5407078 [           0x00000114 nestMembers                  ]  -> 0x00007F58C5407188
     [java] 0x00007F58C5407078-0x00007F58C540707A [               0x0042 majorVersion                 ]
     [java] 0x00007F58C540707A-0x00007F58C540707C [               0x0000 minorVersion                 ]
     [java] 0x00007F58C540707C-0x00007F58C5407080 [           0x00000001 optionalFlags                ]
     [java] 0x00007F58C5407080-0x00007F58C5407084 [           0x000003EC optionalInfo                 ]  -> 0x00007F58C540746C
     [java] 0x00007F58C5407084-0x00007F58C5407088 [           0x00000003 maxBranchCount               ]
     [java] 0x00007F58C5407088-0x00007F58C540708C [           0x00000000 invokeCacheCount             ]
     [java] 0x00007F58C540708C-0x00007F58C5407090 [           0x00000000 bsmCount                     ]
     [java] 0x00007F58C5407090-0x00007F58C5407094 [           0x00000000 callSiteCount                ]
     [java] 0x00007F58C5407094-0x00007F58C5407098 [           0x00000000 callSiteData                 ]
     [java] 0x00007F58C5407098-0x00007F58C540709C [           0x000008E0 classFileSize                ]
     [java] 0x00007F58C540709C-0x00007F58C54070A0 [           0x0000006B classFileCPCount             ]
     [java] 0x00007F58C54070A0-0x00007F58C54070A2 [               0x0000 staticSplitMethodRefCount    ]
     [java] 0x00007F58C54070A2-0x00007F58C54070A4 [               0x0000 specialSplitMethodRefCount   ]
     [java] 0x00007F58C54070A4-0x00007F58C54070A8 [           0x00000000 staticSplitMethodRefIndexes  ]
     [java] 0x00007F58C54070A8-0x00007F58C54070AC [           0x00000000 specialSplitMethodRefIndexes ]
     [java] 0x00007F58C54070AC-0x00007F58C54070B0 [           0x00000000 padding                      ]
     [java]
     [java] === Section End: romHeader                                                                ===
     [java]
     [java] junit.framework.AssertionFailedError
     [java] 	at junit.framework.Assert.fail(Assert.java:48)
     [java] 	at junit.framework.Assert.assertTrue(Assert.java:20)
     [java] 	at junit.framework.Assert.assertTrue(Assert.java:27)
     [java] 	at j9vm.test.ddrext.junit.TestClassExt.testQueryROMClassExt(TestClassExt.java:203)
     [java] 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
     [java] 	at java.base/java.lang.reflect.Method.invoke(Method.java:586)
     [java] 	at junit.framework.TestCase.runTest(TestCase.java:168)
     [java] 	at junit.framework.TestCase.runBare(TestCase.java:134)
     [java] 	at junit.framework.TestResult$1.protect(TestResult.java:110)
     [java] 	at junit.framework.TestResult.runProtected(TestResult.java:128)
     [java] 	at junit.framework.TestResult.run(TestResult.java:113)
     [java] 	at junit.framework.TestCase.run(TestCase.java:124)
     [java] 	at junit.framework.TestSuite.runTest(TestSuite.java:243)
     [java] 	at j9vm.test.ddrext.junit.DDRTestSuite.run(DDRTestSuite.java:74)
     [java] 	at j9vm.test.ddrext.junit.DDRTestSuite.run(DDRTestSuite.java:69)
     [java] 	at j9vm.test.ddrext.junit.DDRTestSuite.run(DDRTestSuite.java:55)
     [java] 	at j9vm.test.ddrext.AutoRun.runTest(AutoRun.java:275)
     [java] 	at j9vm.test.ddrext.AutoRun.main(AutoRun.java:110)
```

Reply:
It seems it just got the name wrong:
```
     [java] 0x00007F58C5407000-0x00007F58C5407004 [           0x00000838 varHandleMethodTypeLookupTable ]  -> 0x00007F58C5407838
```
Looking at the same core file with an older JDK:
```
     [java] 0x00007F58C5407000-0x00007F58C5407004 [           0x00000838 romSize                        ]  -> 0x00007F58C5407838
```
I don't have an explanation, though, because `RomClassWalker.allSlotsInVarHandleMethodTypeLookupTableDo()` shouldn't do anything for a dump from JDK17+.

Reply:
The error persists even after removing the implementation for `allSlotsInVarHandleMethodTypeLookupTableDo`.

Reply:
I think the problem is that checks of `FieldDescriptor.isPresent()` are missing from places like `ClassWalker.addObjectsAsSlot()`.